pkg/ext/extension_apiserver.go

```diff
@@ -96,16 +161,44 @@ func NewExtensionAPIServer(wranglerContext *wrangler.Context) (steveserver.Exten
 			}
 			return aslAuthorizer.Authorize(ctx, attrs)
 		}),
+		SNICerts: []dynamiccertificates.SNICertKeyContentProvider{sniProvider},
 	}
-	codecs := serializer.NewCodecFactory(scheme)
+
 	extensionAPIServer, err := steveext.NewExtensionAPIServer(scheme, codecs, extOpts)
 	if err != nil {
 		return nil, fmt.Errorf("new extension API server: %w", err)
 	}
 
 	if err = extstores.InstallStores(extensionAPIServer, scheme); err != nil {
-		return nil, fmt.Errorf("install stores: %w", err)
+		return nil, fmt.Errorf("failed to install install stores: %w", err)
+	}
+
+	service := Service()
+	if _, err := wranglerContext.Core.Service().Create(&service); client.IgnoreAlreadyExists(err) != nil {
```

Comment:
If the resource already exists we should update it to the "newest" value. This will be easier when developing so we don't have to delete the resources by hand (or turn off the feature), we'll be able to just recompile and start Rancher.

We'll need to be able to update anyway because the service definition might change in an update.